app/forms/hub/coalition_form.rb

```diff
@@ -0,0 +1,18 @@
+module Hub
+  class CoalitionForm < Form
+    include ActiveModel::Model
+
+    attr_accessor :coalition, :name, :states
+
+    def save
+      coalition.name = name
+      form_states = states.split(",").map { |abbr| States.key_for_name(abbr) } # states are comma delimited in a string, i.e. "Ohio,California"
+      existing_states = coalition.state_routing_targets.pluck(:state_abbreviation)
+
+      new_states = form_states - existing_states
+      new_states.each { |state| coalition.state_routing_targets.build(state_abbreviation: state, target: @coalition) }
+      coalition.state_routing_targets = coalition.state_routing_targets.select { |t| form_states.include?(t.state_abbreviation) }
+      coalition.save
```

Comment:
Did you try using Active Record's `replace` here? It will delete the ones that arent included in the collection but if they are included in the collection I think it is smart enough to use the existing record. It probably looks a lot like what you're doing in lines 10 - 14.

```
routing_targets = form_states.map { |state| StateRoutingTarget.new(state_abbreviation: state, target: @coalition) }
coalition.state_routing_targets.replace(routing_targets)
```

The tradeoff is I think you'd have to do two db transactions to persist the data -- one for any changes to the name of the coalition and another for changes to the states included.

Reply:
https://apidock.com/rails/ActiveRecord/Associations/CollectionProxy/replace

Reply:
If we don't want to use AR replace, I'd def suggest pulling the state mapping and replacing logic on 10 -15 into a private method in the form since it's a super specific chunk of functionality.